CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/SyntaxNodesFile.swift

```diff
@@ -304,3 +309,27 @@ private func generateSyntaxChildChoices(for child: Child) throws -> EnumDeclSynt
     }
   }
 }
+
+private func generateInitializerDocComment(for node: LayoutNode) -> SwiftSyntax.Trivia {
+  func generateParamDocComment(for child: Child) -> String? {
+    guard let documentation = child.documentation,
+      let firstLine = documentation.split(whereSeparator: \.isNewline).first
+    else {
+      return nil
+    }
+
+    return "   - \(child.varName): \(firstLine)"
+  }
+
+  let paramsDoc = docCommentTrivia(
+    from: ([
+      " - Parameters:",
+      "   - leadingTrivia: The leading ``Trivia`` for this node.",
+    ]
+      + node.children.compactMap(generateParamDocComment)
+      + ["   - trailingTrivia: The trailing ``Trivia`` for this node."])
+      .joined(separator: "\n")
+  )
+
+  return paramsDoc
+}
```

Comment:
My recent tendency has been to implement these kinds of things in the SyntaxSupport module like I did in https://github.com/apple/swift-syntax/pull/1771. There’s no real strict reason for it, I just thought that it made more sense.

Reply:
Oh, yeah, I've seen it. I think it's a really good approach, but I wonder if it's a good fit for this case. 

I've been thinking about it and I believe that if we would implement it within the `SyntaxSupport` module, like `GrammarGenerator` it would be reasonable to make it reusable. This specific function's implementation is highly contextual and assumes several conditions such as the manual addition of `leadingTrivia` and `trailingTrivia` in the `LayoutNode.generateInitializerDeclHeader(useDeprecatedChildName:)` method - it implies that there is always more than one parameter so we don't need to handle cases for a single parameter (instead of ` - Parameters: (...)` in this case, we should generate ` - Parameter {parameter name}: {parameter documentation}`) or no parameters. We also assume that the `LayoutNode` doesn't know about `leadingTrivia` and `trailingTrivia`. Given these assumptions, I don't think it would be a good idea to just move this function to `SyntaxSupport`. It might be more beneficial to create something more generic, like `ParamsDocCommentGenerator`, which could handle all cases and be useful in other contexts. To address the issue of `LayoutNode` not knowing about `leadingTrivia` and `trailingTrivia`, we might need to pass all parameters to `ParamsDocCommentGenerator` in an abstract way, using a tuple or something. 

If we agree this is the right course of action, I could develop something like the `ParamsDocCommentGenerator`. I'd appreciate your thoughts on this matter! 💙

Reply:
Oh, good point with `leadingTrivia` and `trailingTrivia`. I didn’t think about that. In that case I agree with you that this should be in the SyntaxNodesFile.swift file. Just one minor suggestion: I would make this a fileprivate extension on `LayoutNode` instead of a top-level function. It just reads nicer IMO.

Reply:
Sounds good. So maybe if we wanna convert this function into a method in an extension of `LayoutNode` we could mark it as internal and move it to the `LayoutNode+Extensions.swift` file. Method to generate init header - `LayoutNode.generateInitializerDeclHeader(useDeprecatedChildName:)` is already in this file.